lib/driver.js

```diff
@@ -1,407 +1,388 @@
-  pool.on('log', function (level, message, data) {
-    self.emit('connectionLogged', level, message, data);
-    // unrecoverable errors will yield error on execution, so treat these as warnings since they'll be retried
-    // treat everything else as debug information.
-    var logMethod = (level === 'error' || level === 'warning') ? 'warn' : 'debug';
-
-    var metaData = {
-      datastaxLogLevel: level
-    };
-    if (typeof data === 'string') {
-      message += ': ' + data;
-    } else {
-      metaData.data = data;
-    }
-    self.logger[logMethod]('priam.Driver.' + message, metaData);
-  });
-
-  this.emit('connectionOpening', openRequestId);
-  pool.connect(function (err) {
-    if (err) {
-      self.emit('connectionFailed', openRequestId, err);
-      self.logger.error('priam.Driver: Pool Connect Error',
-        { name: err.name, error: err.message, inner: err.innerErrors });
-      if (waitForConnect) {
-        callback(err, pool);
+    let port = null;
+    if (Array.isArray(dsPoolConfig.contactPoints)) {
+      for (let i = 0; i < dsPoolConfig.contactPoints.length; i++) {
+        const split = dsPoolConfig.contactPoints[i].split(':');
+        dsPoolConfig.contactPoints[i] = split[0].trim();
+        if (split.length > 1) {
+          port = parseInt(split[1].trim(), 10);
+        }
+      }
+      if (port !== null) {
+        dsPoolConfig.protocolOptions = dsPoolConfig.protocolOptions || {};
+        dsPoolConfig.protocolOptions.port = port;
       }
-      self.callWaiters(err, pool);
-      return void self.closePool(pool);
     }
-    pool.isReady = true;
-    self.emit('connectionOpened', openRequestId);
-    if (waitForConnect) {
-      callback(null, pool);
+    if (poolConfig.getAConnectionTimeout) {
+      dsPoolConfig.socketOptions = dsPoolConfig.socketOptions || {};
+      dsPoolConfig.socketOptions.connectTimeout = poolConfig.getAConnectionTimeout;
     }
-    self.callWaiters(null, pool);
-  });
-  if (!waitForConnect) {
-    callback(null, pool);
+    if (poolConfig.poolSize) {
+      dsPoolConfig.pooling = dsPoolConfig.pooling || {};
+      dsPoolConfig.pooling.coreConnectionsPerHost = dsPoolConfig.pooling.coreConnectionsPerHost || {};
+      dsPoolConfig.pooling.coreConnectionsPerHost[cqlDriver.types.distance.local.toString()] = poolConfig.poolSize;
+      dsPoolConfig.pooling.coreConnectionsPerHost[cqlDriver.types.distance.remote.toString()] = Math.ceil(poolConfig.poolSize / 2);
+    }
+    return dsPoolConfig;
   }
-};
-
-DatastaxDriver.prototype.remapConnectionOptions = function remapConnectionOptions(connectionData) {
-  remapOption(connectionData, 'user', 'username');
-  remapOption(connectionData, 'hosts', 'contactPoints');
-};
 
-DatastaxDriver.prototype.connect = function connect(keyspace, callback) {
-  if (typeof keyspace === 'function') {
-    callback = keyspace;
-    keyspace = null;
+  _remapConnectionOptions(connectionData) {
+    remapOption(connectionData, 'user', 'username');
```

Comment:
This remapping is also from the old `helenus` and `node-cassandra-cql` days. I'm fine keeping our same interface, but just wanted to give you the option of removing this code to make it closer to the official driver interface.

Reply:
@scommisso: done